Clustering/src/KMeans.java

```diff
@@ -1,10 +1,117 @@
 /**
  * K-Means clustering
  */
+import java.lang.Math;
+import java.util.ArrayList;
+
 public class KMeans  extends Clustering{
 
+    /**
+     * Clusters data via the K-means Algorithm
+     */
     @Override
     public int[] cluster(double[][] data, int numClusters){
+        int cycles =0 ;
+        double[][] centroids = initCentroids(data, numClusters);
+        while(!hasConverged(cycles)){
+            ArrayList<Integer> labels = getLabels(centroids, data);
+            centroids = getCentroids(labels, data, numClusters);
+            cycles++;
+        }
         return null;
     }
+
+    /**
+     * returns true if K-means has converged
+     */
+    private boolean hasConverged(int cycles) {
+        return (cycles >= 200);
+    }
```

Comment:
If this is the only convergence criteria, I don't we think it needs its own method

Reply:
true